lib/internal/fs/recursive_watch.js

```diff
@@ -155,10 +155,10 @@ class FSWatcher extends EventEmitter {
           }
 
           if (file.isFile()) {
+            this.#files.set(f, file);
```

Comment:
We don't need this, `watchFile` already sets it.

Reply:
it only sets it on the first change, if no changes this will cause a leakage where open `StatWatchers` exists and `.close` wont close them

Reply:
Nice catch! Can you add a comment?

Reply:
will do 👍🏻 I am exploring various use cases running `node --watch` and `node --test --watch` - this is what led me to these edge cases so I want to make sure everything runs ok before I fix all comments and separate to a new PR